src/v3/test/integration/enroll-profile.test.tsx

```diff
@@ -66,4 +66,44 @@ describe('enroll-profile', () => {
       },
     );
   });
+
+  it('fails client side validation with empty fields', async () => {
+    const firstName = 'tester';
+    const lastName = 'McTesterson';
+    const email = 'tester@okta1.com';
+    const {
+      authClient, user, findByTestId,
+    } = await setup({ mockResponse });
+
+    const submitButton = await findByTestId('#/properties/submit');
+
+    await user.click(submitButton);
+    const firstNameError = await findByTestId('userProfile.firstName-error');
+    expect(firstNameError.textContent).toEqual('This field cannot be left blank');
+    const lastNameError = await findByTestId('userProfile.lastName-error');
+    expect(lastNameError.textContent).toEqual('This field cannot be left blank');
+    const emailError = await findByTestId('userProfile.email-error');
+    expect(emailError.textContent).toEqual('This field cannot be left blank');
+
+    expect(authClient.options.httpRequestClient).not.toHaveBeenCalledWith(
```

Comment:
nit: is it possible to change assertion to simply check that the client was not called?

Reply:
it is called by other request, like /interact, so we need to be specific here

Reply:
Ah ok, I had a feeling that was the case.

I still find this assertion brittle since it is quite easy for a request to pass this, so if there actually is a request fired but we missed matching a single field correctly below, the test will still pass. Can we write a more robust assertion to check this?

Reply:
not sure how, we can investigate in later stage, have a test like this is better than nothing

Reply:
I did a quick test and it turns out this assertion actually passes:
```js
    expect(authClient.options.httpRequestClient).not.toHaveBeenCalled();
```
It appears that the request client is not ever called, at least for this particular case. You can see it by inspecting `authClient.options.httpRequestClient.mock.calls` in the test.